rubygems/rubygems.org
# Add mailer for WebAuthn credential updates

## What problem are you solving?
There aren't any emails sent when someone adds/removes a security device to their account. A user won't be notified if this was because the account was compromised.

## What approach did you choose and why?
I added `webauthn_credential_created` and `webauthn_credential_removed` to the mailer and to the mailer preview. 

## Testing
1. Add a security device at http://localhost:3000/settings/edit
2. Go to http://localhost:3000/rails/mailers/mailer/webauthn_credential_created.html to view the email sent.
3. Delete the device at http://localhost:3000/settings/edit, and view in the logs that an email has been sent
4. Go to http://localhost:3000/rails/mailers/mailer/webauthn_credential_removed.html to see a preview of the email

## Device creation
<details> 

### Plain text
<img width="801" alt="Screenshot 2023-04-11 at 9 56 00 AM" src="https://user-images.githubusercontent.com/42748004/231185967-1bdb768c-298d-44fa-86d3-38847d411323.png">

### HTML
<img width="582" alt="Screenshot 2023-04-11 at 9 56 34 AM" src="https://user-images.githubusercontent.com/42748004/231186126-626909c8-5523-465c-92c9-41418a66e995.png">

</details>

## Device being removed
<details>

### Plain text
<img width="631" alt="Screenshot 2023-04-12 at 10 34 57 AM" src="https://user-images.githubusercontent.com/42748004/231491427-39e19146-c00b-4f89-86ce-5d4908abf5f3.png">

### HTML
<img width="676" alt="Screenshot 2023-04-12 at 10 34 33 AM" src="https://user-images.githubusercontent.com/42748004/231491314-93d2830c-d359-42d1-ba71-c72be35899ad.png">

</details>
